src/Command/Util/ApigeeEdgeManagementCliService.php

```diff
@@ -0,0 +1,330 @@
+   *
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The HTTP client.
+   */
+  public function __construct(ClientInterface $http_client) {
+    $this->httpClient = $http_client;
+  }
+
+  /**
+   * Create role in Apigee Edge for Drupal to use for Edge connection.
+   *
+   * @param \Symfony\Component\Console\Style\StyleInterface $io
+   *   The IO interface of the CLI tool calling the method.
+   * @param callable $t
+   *   The translation function akin to t().
+   * @param string $org
+   *   The organization to connect to.
+   * @param string $email
+   *   The email of an Edge user with org admin role to make Edge API calls.
+   * @param string $password
+   *   The password of an Edge user with org admin role to make Edge API calls.
+   * @param null|string $base_url
+   *   The base url of the Edge API.
+   * @param null|string $role_name
+   *   The role name to add the permissions to.
+   */
+  public function createEdgeRoleForDrupal(StyleInterface $io,
+                                          callable $t,
+                                          string $org,
+                                          string $email,
+                                          string $password,
+                                          ?string $base_url,
+                                          ?string $role_name) {
+
+    if ($base_url === NULL || $base_url === '') {
+      // Set default base URL if var is null or empty string.
+      $base_url = self::DEFAULT_BASE_URL;
+    }
+    else {
+      // Validate it is a valid URL.
+      if (!UrlHelper::isValid($base_url, TRUE)) {
+        $io->error($t('Base URL is not valid.'));
+        return;
+      }
+
+    }
+
+    if ($role_name === NULL || $role_name === '') {
+      // Set default if null or empty string.
+      $role_name = self::DEFAULT_ROLE_NAME;
+    }
+
+    if (!$this->isValidEdgeCredentials($io, $t, $org, $email, $password, $base_url)) {
+      return;
+    }
+
+    $does_role_exist = $this->doesRoleExist($org, $email, $password, $base_url, $role_name);
+
+    // If role does not exist, create it.
+    if (!$does_role_exist) {
+      $io->text($t('Role :role does not exist. Creating role.', [':role' => $role_name]));
+
+      $url = "{$base_url}/o/{$org}/userroles";
+      try {
+        $this->httpClient->post($url, [
+          'body' => json_encode([
+            'role' => [$role_name],
+          ]),
+          'auth' => [$email, $password],
+          'headers' => [
+            'Accept' => 'application/json',
+            'Content-Type' => 'application/json',
+          ],
+        ]);
+      }
+      catch (TransferException $exception) {
+        $this->handleHttpClientExceptions($exception, $io, $t, $url, $org, $email);
+        return;
+      }
+    }
+    else {
+      $io->text('Role ' . $role_name . ' already exists.');
+    }
+
+    $this->setDefaultPermissions($io, $t, $org, $email, $password, $base_url, $role_name);
```

Comment:
This could potentially reset some permissions if the role already exists and had its permissions modified from default. Maybe only set defaults if the role was just created? ie move into `try {}` block

Reply:
Also per the docs, this call "does not remove any existing permissions in the role", but I agree with @kscheirer that calling it for an existing role (which might have been altered outside of this call) might create the  impression that this command validated the role to only have the required permissions. I think that a) either exit with error if the role already exists, or b) force resetting the role, which would involve deleting all permissions and adding the right ones again.

Reply:
Was thinking this was OK, but you both bring up points that someone may make changes to the role so we do not want to stomp those changes.

Reply:
Added a `--force` flag